### DISCUSSION TITLE
Request for adding CodeQL to winget

### DISCUSSION THREAD

**Discussion by Shion1305** [github/codeql#14296] [Ideas]
Hello everyone,

First off, I apologize for any mistakes in English - it's not my native language.

I want CodeQl to be available on winget(Windows Package Manager).

I'm very new to CodeQL. I've always relied on winget to manage tool installations on Windows machines, finding it incredibly convenient. Recently, I was introduced to CodeQL in my university class, and was disappointed to discover that it's not available through winget. Installation by extracting from a zip file feels cumbersome, especially when managing and updating tools. Today, many applications are available via winget, and having CodeQL be one of them would significantly enhance its accessibility and user experience.

Adding CodeQL to winget would not only simplify the process for newcomers like myself but also provide a consistent experience alongside other tools. I hope others share this sentiment and we can see CodeQL on winget in the near future.

**Comment by aibaars** [comment]
Thanks for you question

> First off, I apologize for any mistakes in English - it's not my native language.

No worries. Your English is very good.

It's a good idea, but I don't think there are any plans for publishing CodeQL via package managers of the various operating systems we support. I'll pass it on to the people that are responsible for CodeQL releases.

For managing CodeQL and switching versions I use [`gh codeql`](https://github.com/github/gh-codeql#readme) myself. It is an extension of the GithHub CLI. This is available on winget, see also https://github.com/cli/cli#installation . I'm not sure if the `gh codeql` extension works on windows though, but it should be easy enough to try:
* get the `gh` cli from winget
* run `gh extensions install github/gh-codeql`
* run `gh codeql`
